test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -0,0 +1,412 @@
+        if len(sku_installed) < 2:
+            return False
+        else:
+            return True
+
+    def _process_message(
+            self,
+            action,
+            typeid,
+            nodeid,
+            messagetype,
+            amqp_message_body):
+        expected_key = messagetype + "." + action + \
+            ".information." + typeid + "." + nodeid
+        expected_payload = {
+            "type": messagetype,
+            "action": action,
+            "typeId": typeid,
+            "nodeId": nodeid,
+            "severity": "information",
+            "version": "1.0",
+            "createdAt": {}}
+        self._compare_message(
+            amqp_message_body,
+            expected_key,
+            expected_payload)
+
+    def _compare_message(self, amqpmessage, expected_key, expected_payload):
+        routing_key = amqpmessage[0]
+        amqp_body = amqpmessage[2]
+        self.assertEquals(
+            routing_key,
+            expected_key,
+            "Routing key is not expected! expect {0}, get {1}" .format(
+                expected_key,
+                routing_key))
+        try:
+            amqp_body_json = fit_common.json.loads(amqp_body)
+        except ValueError:
+            logs.error("FAILURE - The message body is not json format!")
+            return
+        try:
+            self.assertEquals(
+                amqp_body_json['version'],
+                expected_payload['version'],
+                "version field not correct! expect {0}, get {1}" .format(
+                    expected_payload['version'],
+                    amqp_body_json['version']))
+            self.assertEquals(
+                amqp_body_json['typeId'], expected_payload['typeId'],
+                "typeId field not correct!  expect {0}, get {1}"
+                .format(expected_payload['typeId'], amqp_body_json['typeId']))
+            self.assertEquals(
+                amqp_body_json['action'], expected_payload['action'],
+                "action field not correct!  expect {0}, get {1}"
+                .format(expected_payload['action'], amqp_body_json['action']))
+            self.assertEquals(
+                amqp_body_json['severity'],
+                expected_payload['severity'],
+                "serverity field not correct!" .format(
+                    expected_payload['severity'],
+                    amqp_body_json['severity']))
+            self.assertNotEquals(
+                amqp_body_json['createdAt'],
+                {},
+                "createdAt field is empty!")
+            self.assertNotEquals(
+                amqp_body_json['data'],
+                {},
+                "data field is empty!")
+        except ValueError as e:
+            self.fail(
+                "FAILURE - expected key is missing in the AMQP message!{0}".format(e))
+
+    def test_rediscover(self):
+        node_collection = test_api_utils.get_node_list_by_type("compute")

```

Comment:
Add one line like this: "asssert node_collection", in case we get an empty node_collection and the case is not running at all.